### PR TITLE
Feat: Added support for Ed25519Signature2020 signature type and Ed25519VerificationKey2020

### PR DESCRIPTION
**Pull request by dkulic**

### CODE REVIEW

File: aries_cloudagent/vc/ld_proofs/suites/ed25519_signature_2020.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            proof (dict): The proof to add the signature to
+
+        Returns:
+            dict: The proof object with the added signature
+
+        """
+        signature = await self.key_pair.sign(verify_data)
+        pprint("Signature")
+        pprint(signature)
+        pprint(len(signature))
```
Comment (TimoGlastra) [inline aries_cloudagent/vc/ld_proofs/suites/ed25519_signature_2020.py:60]:
```suggestion
```

**Review by TimoGlastra — APPROVED** [review]
LGTM, some primt statements. There are quite some places where suites are registered. Might be good to add a registry at some point for signature suites, which would also allow for dynamic registration of more suites

**Comment by dkulic** [comment]
I will fix these errors, sorry for that, but one of them is multiformats library not supporting python 3.6. Python 3.6 is EOL, maybe it should be dropped?

**Comment by WadeBarnes** [comment]
@swcurran, Yes we can drop Python 3.6.  The `py3.9` images have shown to be drop-in compatible where the BC Gov Python 3.6 based images were used before in all cases where I've upgraded to `0.8.1`.

**Comment by dbluhm** [comment]
I opened up #2247 to make the version bump from 3.6 to 3.9

**Comment by swcurran** [comment]
To be discussed at ACA-Pug Meeting — May 30, 2023.

**Comment by swcurran** [comment]
Per the May 30, 2023 ACA-Pug meeting — we are dropping 3.6 support in ACA-Py #2244 / #2247 . As soon as that is done, this should be updated, processed and merged.

**Comment by dbluhm** [comment]
@dkulic is OOTO so I'll take care of the conflicts on this PR.

**Review by dbluhm — APPROVED** [review]

**Comment by dbluhm** [comment]
Requesting input from @usingtechnology and/or @andrewwhitehead; would like to see this one keep moving along.

File: aries_cloudagent/connections/base_manager.py [hunk 2]
```diff
@@ -294,6 +299,8 @@ async def resolve_invitation(
     def _extract_key_material_in_base58_format(method: VerificationMethod) -> str:
         if isinstance(method, Ed25519VerificationKey2018):
             return method.material
+        elif isinstance(method, Ed25519VerificationKey2020):
+            return bytes_to_b58(multibase.decode(method.material))
```
Comment (dbluhm) [inline aries_cloudagent/connections/base_manager.py:303]: I think this also needs an "unwrap" on the material to remove the `0xed` multicodec prefix added when using the `publicKeyMultibase`

**Review by dbluhm — CHANGES_REQUESTED** [review]

File: aries_cloudagent/connections/base_manager.py (hunk 2 above)
Comment (dkulic) [reply aries_cloudagent/connections/base_manager.py]: I added support for both (with and without multicodec). Do you think it should be like that or should we support multibase keys only with multicodec?

Comment (dbluhm) [reply aries_cloudagent/connections/base_manager.py]: Given the ambiguity, I think accepting either is probably a good choice for now at least.

**Review by dbluhm — APPROVED** [review]

**Pull request merged**